### PR TITLE
6/29 triage annotations

### PR DESCRIPTION
**Pull request by psahabu**
- XC reverting to gcc 6.3
- new tmp dir
- less locking on associative arrays
- regex leak improvements
- updateFluff fix

### CODE REVIEW

File: test/ANNOTATIONS.yaml [hunk 1]
```diff
@@ -472,6 +481,8 @@ hpl_performance:
 hpl.hpcc2012.ml-perf:
   03/03/17:
     - Modified test as part of array views (#5338)
+  06/19/17:
+    - Reduce amount of locking in associative array dsiAccess (#6244)
```
Comment (benharsh) [inline test/ANNOTATIONS.yaml]: Did you verify this was the cause?

File: test/ANNOTATIONS.yaml [hunk 2]
```diff
@@ -769,14 +788,10 @@ prk-stencil.ml-perf:
...
   06/06/17:
     - Increase the PRK stencil problem size (#6380)
-  06/13/17:
-    - Optimize updateFluff (#6418)
 
 prk-stencil-time:
   05/11/17:
     - Move chpl_getPrivatizedClass() into chpl-privatization.h (#6212)
-  06/13/17:
-    - Optimize updateFluff (#6418)
```
Comment (benharsh) [inline test/ANNOTATIONS.yaml]: Why remove these?

Comment (psahabu) [inline test/ANNOTATIONS.yaml]: I messed up the first time around, they didn't actually do anything to the stencil as far as I could tell. I can put them back in though.

File: test/ANNOTATIONS.yaml (hunk 1 above)
Comment (psahabu) [inline test/ANNOTATIONS.yaml]: I didn't verify. Since it was a positive change, should I just remove it?

**Pull request merged**